### ISSUE TITLE
BUG: core dumps when calling scipy.interpolate.interp1d with array of nans

### ISSUE THREAD

**Issue by jfmoulin** [scipy/scipy#11637]
I would expect  error to be raised rather than allowing execution of such a show stopper, even after a warning ... 

#### Reproducing code example:
```
jfmoulin@jfmlap:~$ ipython --matplotlib=qt
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.13.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: from scipy.interpolate import interp1d                                  

In [2]: import numpy as np                                                      

In [3]: x = np.ones(10) * np.nan                                                

In [4]: y = p.rand(10)                                                          

In [5]: interpolant = interp1d(x,y,kind=3)                                      
```

#### Error message:
/usr/local/lib/python3.6/dist-packages/scipy/interpolate/_bsplines.py:798: RuntimeWarning: invalid value encountered in less_equal
  if x.ndim != 1 or np.any(x[1:] <= x[:-1]):
/usr/local/lib/python3.6/dist-packages/scipy/interpolate/_bsplines.py:802: RuntimeWarning: invalid value encountered in less
  if t.ndim != 1 or np.any(t[1:] < t[:-1]):

In [6]: double free or corruption (!prev)                                       
Aborted (core dumped)
#### Scipy/Numpy/Python version information:
```
In [1]: import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sy
   ...: s.version_info)                                                         
1.4.1 1.14.5 sys.version_info(major=3, minor=6, micro=9, releaselevel='final', serial=0)
```

**Comment by ev-br** [comment]
Indeed.  gh-11639 should fix it.

**Comment by jfmoulin** [comment]
Hi!
Thanks for the fast feedback and action. 
I now found some time to test. and it passes ! I can cleanup my try/except :0)

Just a practical question:
My scipy was pip3 installed.
I edited the file by hand, but in general, how should I proceed to test such a fix ?
 How should I patch the file ?  Should I rather install this branch ? How (I usually only install master...)?
I would happily rtfm if you point me to it !

Thanks again for the great work with scipy/numpy.
